.github/workflows/pythonapp.yml

```diff
@@ -0,0 +1,31 @@
+name: Python application
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
Maybe all these options can be moved into a config file (setup.cfg, tox.ini, etc)?
CI is good, but local linting and testing (without pushing to the repo) is much quicker (and thus better). Also configs can be picked up by other tools like language servers.
I'd prefer to have linting and testing as `tox` actions (and then just run `tox` on github actions)

Reply:
I agree that CI should reuse tool configuration as much as possible in order to not repeat oneself. Flake8 enables this by reading various config files in search of its section:

http://flake8.pycqa.org/en/2.6.0/config.html#per-project

However, a tox configuration is still missing for this project. Since that seems like that would have larger implications, I would like to handle it at a later date. Since a setup.cfg already existed, I added the command line flags to it. These should be reusable for later tox integration.